lib/invoice_item_repository.rb

```diff
@@ -0,0 +1,69 @@
+class InvoiceItemRepository
+  attr_reader :invoice_items
+
+  def initialize(filename)
+    @invoice_items = create_invoice_items(filename)
+  end
+
+  def create_invoice_items
+    FileIo.process_csv(filename, InvoiceItem)
+  end
+
+  def all
+    @invoice_items
+  end
+
+  def find_by_id(id)
+    @invoice_items.find do |invoice_item|
+      invoice_item.id == id
+    end
+  end
+
+  def find_all_by_item_id(id)
+    @invoice_items.select do |invoice_item|
+      invoice_item.item_id == id
+    end
+  end
+
+  def find_all_by_invoice_id(id)
+    @invoice_items.select do |invoice_item|
+      invoice_item.invoice_id == id
+    end
+  end
+
+  def create_first_invoice_item(attributes)
+    new_invoice_item = InvoiceItem.new(attributes)
+    new_invoice_item.update_id(1)
+    @invoice_items << new_invoice_item
+  end
```

Comment:
Looks good! I think we're a bit inconsistent with this across classes -- not a problem here, but maybe when we deal with inheritance we can ensure all the repo classes are in line.

Reply:
(Don't think we have this guard in Merchant class and I'm not totally sure we need it anywhere)